### DISCUSSION TITLE
1:20200101.8 is latest?

### DISCUSSION THREAD

**Discussion by philipmw** [solokeys/solo2#89] [Q&A] [answered]
Hi,

When I run `solo2 update` with debugging enabled, the tool downloads `solo2-firmware-1.0.8.sb2` from this repo, but then offers to install firmware `1:20200101.8` onto the key. I am surprised at this firmware, as that's dated almost two years ago, if reading it as a YYYYMMDD date, while GitHub shows that firmware 1.0.8 was created just 25 days ago.

Is it expected that solo2 is trying to install firmware `1:20200101.8` as the latest?

**Comment by nickray** [comment, marked as answer]
Yes, it's the latest.

An "X.Y.Z" version can equivalently be as "1:date.Z", where date = 2020-01-01 + Y days.
The hope is that there won't be too many breaking changes (X, "era"), just functionality improvements and bugfixes (Y, "date").
So then the date component gives you insight into how old your firmware is, and whether you might want to look for an update.
And the patch component Z would only be needed for intra-day mis-release fixing.

Unfortunately the first few versions were released with Y = 0, which then translates into Jan 1 of 2020.

The next release should hopefully use this calver choice of Y.
